tests/unittests/sources/test_wsl.py

```diff
@@ -469,14 +560,175 @@ def test_data_precedence(self, m_get_linux_dist, tmpdir, paths):
             ),
         )
 
-        assert "wsl.conf" not in userdata
-        assert "packages" not in userdata
-        assert "ubuntu_advantage" in userdata
+        assert "ubuntu_pro" in userdata, "Agent data should be present"
         assert "package_update" in userdata, (
             "package_update entry should not be overriden by agent data"
             " nor ignored"
         )
-        assert "landscape" in userdata
         assert (
             "landscapetest" not in userdata and "agenttest" in userdata
         ), "Landscape account name should have been overriden by agent data"
+        # Make sure we have tags from Landscape data, not agent's
+        assert (
+            "tag_aiml" in userdata and "tag_dev" in userdata
+        ), "User-data should override agent data's Landscape computer tags"
```

Comment:
Let's strictly assert we don't see `wsl` as well from agent.yaml. Also note landscape client config schema disallows spaces between tags. so our example should be `tags: x,y`
```suggestion
        assert (
            "tags: tag_aiml,tag_dev" in userdata
        ), "User-data should override agent data's Landscape computer tags"
        assert "wsl" not in userdata
```